test/distributed/test_device_mesh.py

```diff
@@ -526,6 +526,29 @@ def test_cache_and_reuse_submesh_slice_result(self):
         tp_mesh = mesh["tp"]
         self.assertEqual(_world.group_count, ref_pg_count)
 
+    @with_comms
+    def test_get_item_3d_noncontinuous_slicing(self):
+        mesh_shape = (2, 2, 2)
+        mesh_dim_names = ("dp", "pp", "cp")
+        mesh_3d = init_device_mesh(
+            self.device_type, mesh_shape, mesh_dim_names=mesh_dim_names
+        )
+
+        # Slice order simply decides which mesh_dim sits on which mesh_dim.
+        # For dp_cp_mesh, cp mesh is the innermost dimension.
+        dp_cp_mesh = mesh_3d["dp", "cp"]
+        cp_mesh = mesh_3d["cp"]
+        dp_local_rank = dp_cp_mesh.get_local_rank("dp")
+        # Check on the current dp_local_rank, whether the cp mesh tensor is the same.
+        self.assertEqual(dp_cp_mesh.mesh[dp_local_rank], cp_mesh.mesh)
```

Comment:
In addition to this check, can you directly check if the dp_cp_mesh tensor match with the expected mesh tensor we want?